common/txmgr/types/client.go

```diff
@@ -56,3 +59,48 @@ type Blocks[BLOCK any, BLOCKHASH types.Hashable] interface {
 type Events[EVENT any, EVENTOPS any] interface {
 	FilterEvents(ctx context.Context, query EVENTOPS) ([]EVENT, error)
 }
+
+type TxmClient[
+	CHAIN_ID ID,
+	HEAD Head,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	R ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ Sequence,
+	FEE Fee,
+	ADD any,
+] interface {
+	BatchSendTransactions(
+		ctx context.Context,
+		store TxStore[ADDR, CHAIN_ID, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD],
+		attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		bathSize int,
+		lggr logger.Logger,
+	) ([]clienttypes.SendTxReturnCode, []error, error)
+	SendTransactionReturnCode(
+		ctx context.Context,
+		tx Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+		lggr logger.Logger,
+	) (clienttypes.SendTxReturnCode, error)
+	PendingNonceAt(ctx context.Context, addr ADDR) (int64, error)
+	SequenceAt(ctx context.Context, addr ADDR, blockNum *big.Int) (SEQ, error)
+	BatchGetReceipts(
+		ctx context.Context,
+		attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD],
+	) (txReceipt []R, txErr []error, err error)
+	SendEmptyTransaction(
+		ctx context.Context,
+		txAttemptBuilder TxAttemptBuilder[HEAD, FEE, ADDR, TX_HASH, Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD], TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD], SEQ],
```

Comment:
Does it make sense to keep going for these diabolic cases?
```suggestion
		txAttemptBuilder TxAttemptBuilder[
			HEAD, FEE, ADDR, TX_HASH, 
			Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD], 
			TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD], 
			SEQ,
		],
```

Reply:
@jmank88 cleaned up TxAttemptBuilder so it doesn't use the Tx/TxAttempt types in the declaration - is that what you were referring to?

Reply:
Yeah, looks good now w/o any excess nesting.